In [ ]:
#Import pandas
import pandas as pd

In [ ]:
#Load the dataset
df=pd.read_csv(r"C:\Users\marya\OneDrive\Desktop\fintech\finpay-data-analytics-roadmap\pandas\customers_features.csv")

,customer_id,first_name,last_name,gender,age,city,country,signup_date,employment_status,annual_income,...,signup_quarter,customer_tenure_days,monthly_income,high_value_customer,annual_income_k,risk_score,age_group,income_band,credit_score_category,risk_segment
0,C001,Sara,Khan,Female,35,Jeddah,Saudi Arabia,2024-03-12,Student,66868.0,...,1,854,5572.333333,No,66.9,3,Adult,Medium,Excellent,High Risk
1,C002,Maryam,Iqbal,Female,22,Riyadh,Saudi Arabia,2024-02-17,Salaried,100990.0,...,1,878,8415.833333,Yes,101.0,3,Young Adult,High,Excellent,High Risk
2,C003,Ali,Qureshi,Male,54,Khobar,Saudi Arabia,2024-04-22,Self-Employed,112926.0,...,2,813,9410.500000,No,112.9,1,Middle Aged,High,Poor,Safe
3,C004,Ahmed,Malik,Female,29,Jeddah,Saudi Arabia,2025-05-05,Self-Employed,66793.0,...,2,435,5566.083333,No,66.8,2,Adult,Medium,Poor,Monitor
4,C005,Sara,Malik,Female,58,Dammam,Saudi Arabia,2025-02-17,Salaried,102066.5,...,1,512,8505.541667,Yes,102.1,1,Middle Aged,High,Excellent,Safe
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
92,C096,Omar,Farooq,Female,37,Jeddah,Saudi Arabia,2024-10-26,Self-Employed,168902.0,...,4,626,14075.166667,No,168.9,2,Adult,High,Poor,Monitor
93,C097,Fatima,Hussain,Female,40,Riyadh,Saudi Arabia,2025-03-15,Salaried,126456.0,...,1,486,10538.000000,No,126.5,3,Middle Aged,High,Good,High Risk
94,C098,Lina,Shaikh,Female,55,Riyadh,Saudi Arabia,2024-08-20,Salaried,122448.0,...,3,693,10204.000000,No,122.4,2,Middle Aged,High,Fair,Monitor
95,C099,Sara,Hussain,Male,54,Khobar,Saudi Arabia,2024-07-30,Salaried,89173.0,...,3,714,7431.083333,No,89.2,2,Middle Aged,Medium,Excellent,Monitor


In [ ]:
#Display the dataframe
df

In [4]:
#Check dataset information
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 97 entries, 0 to 96
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   customer_id            97 non-null     str    
 1   first_name             97 non-null     str    
 2   last_name              97 non-null     str    
 3   gender                 97 non-null     str    
 4   age                    97 non-null     int64  
 5   city                   97 non-null     str    
 6   country                97 non-null     str    
 7   signup_date            97 non-null     str    
 8   employment_status      97 non-null     str    
 9   annual_income          97 non-null     float64
 10  credit_score           97 non-null     float64
 11  risk_level             97 non-null     str    
 12  signup_year            97 non-null     int64  
 13  signup_month           97 non-null     str    
 14  signup_day             97 non-null     int64  
 15  signup_weekday     

In [5]:
#Check missing values
df.isnull().sum()

customer_id              0
first_name               0
last_name                0
gender                   0
age                      0
city                     0
country                  0
signup_date              0
employment_status        0
annual_income            0
credit_score             0
risk_level               0
signup_year              0
signup_month             0
signup_day               0
signup_weekday           0
signup_quarter           0
customer_tenure_days     0
monthly_income           0
high_value_customer      0
annual_income_k          0
risk_score               0
age_group                0
income_band              0
credit_score_category    0
risk_segment             0
dtype: int64

In [6]:
#Check duplicated rows
df.duplicated().sum()

np.int64(0)

In [7]:
#Check dataset dimensions
df.shape

(97, 26)

In [8]:
#Total customers
len(df)

97

In [11]:
#Average Customer Age
df["age"].mean().round(2)

np.float64(41.08)

In [12]:
#Average Annual Income
df["annual_income"].mean().round(2)

np.float64(107775.27)

In [ ]:
#Average Credit Score 
df["credit_score"].mean().round(2)

np.float64(688.34)

In [17]:
#Percentage of High Risk Customers
(df["risk_segment"]=="High Risk").mean()*100

np.float64(29.896907216494846)

In [18]:
#Which city has the largest customer base?
df["city"].value_counts()

city
Riyadh    25
Khobar    21
Jeddah    18
Dammam    18
Mecca     15
Name: count, dtype: int64

In [19]:
#Which employment status is most common?
df["employment_status"].value_counts()

employment_status
Self-Employed    34
Student          31
Salaried         29
Unknown           3
Name: count, dtype: int64

In [20]:
#Which age group has the most customers?
df["age_group"].value_counts()

age_group
Middle Aged    45
Adult          40
Young Adult     8
Senior          4
Name: count, dtype: int64

In [21]:
#Which city has the highest average annual income?
df.groupby("city")["annual_income"].mean().sort_values(ascending=False)

city
Jeddah    115676.055556
Khobar    115207.952381
Dammam    109768.694444
Riyadh    100499.500000
Mecca      97622.733333
Name: annual_income, dtype: float64

In [22]:
#Which income band contains the most customers?
df["income_band"].value_counts()

income_band
High      51
Medium    40
Low        6
Name: count, dtype: int64

In [24]:
#Which employment status has the highest average income?
df.groupby("employment_status")["annual_income"].mean().sort_values(ascending=False)

employment_status
Self-Employed    113105.941176
Salaried         107040.844828
Unknown          105091.000000
Student          102875.532258
Name: annual_income, dtype: float64

In [25]:
#Which city has the highest average credit score?
df.groupby("city")["credit_score"].mean().sort_values(ascending=False)

city
Riyadh    715.600000
Khobar    714.523810
Dammam    675.666667
Mecca     666.933333
Jeddah    650.444444
Name: credit_score, dtype: float64

In [26]:
#How many customers belong to each risk segment?
df["risk_segment"].value_counts()

risk_segment
Safe         38
Monitor      30
High Risk    29
Name: count, dtype: int64

In [28]:
#Which city has the most high risk customers?
df[df["risk_segment"]=="High Risk"]["city"].value_counts()

city
Khobar    9
Jeddah    6
Riyadh    5
Dammam    5
Mecca     4
Name: count, dtype: int64

In [30]:
#How many high-value customers are there?
df["high_value_customer"].value_counts()

high_value_customer
No     81
Yes    16
Name: count, dtype: int64

In [34]:
#Which city has the most high-value customers?
df[df["high_value_customer"]=="Yes"]["city"].value_counts()

city
Riyadh    5
Dammam    3
Jeddah    3
Mecca     3
Khobar    2
Name: count, dtype: int64

In [35]:
#Which month had the most customer signups?
df["signup_quarter"].value_counts()

signup_quarter
1    38
2    33
4    15
3    11
Name: count, dtype: int64

In [37]:
#What is the average customer tenure?
df["customer_tenure_days"].mean().round(2)

np.float64(686.92)

Business Summary

- The dataset contains 97 customer records
- Riyadh has the largest customer base
- The average customer age is 41 years
- The average annual income is SAR 107775
- The average credit score is 688
- Medium Income is the largest customer segment
- Middle Aged represent the largest age group
- Self-Employed customers have the highest average annual income
- Q1 recorded the highest number of customer signups 

Business Recommendations

- Focus on customer acquisition campaigns in Riyadh, as it has the largest customer base
- Offer premium financial products and higher credit limits to self-employed customers, who have the highest annual income.
- Develop targeted financial products and marketing campaigns for middle aged customers, the company's largest customer segment
- Design affordable products and loyalty programs for medium-income customers to improve engagement and long term retention
- Analyze the factors contributing to strong Q1 customer acquisition and apply successful strategies across remaining quarters 
- Continue monitoring customer credit profiles to support responsible lending and maintain portfolio quality. 